app/src/main/java/org/mozilla/reference/browser/browser/BrowserFragment.kt

```diff
@@ -36,6 +38,10 @@ class BrowserFragment : Fragment(), BackHandler, DownloadDialogListener {
 
         val sessionId = arguments?.getString(SESSION_ID)
 
+        historyTrackingFeature = HistoryTrackingFeature(
+                requireComponents.engine,
+                requireComponents.historyStorage)
```

Comment:
I assume this is more like an "application level" feature and we would want to create this in `BrowserApplication.onCreate()` so that we track history even when there's no fragment/UI around. (cc @grigoryk)

Reply:
I'm not sure what "tracking history without a browser fragment/ui around" means, but generally I agree, history tracking isn't conceptually tied to the UI, so moving this over to `BrowserApplication` makes sense.

Separately, we'd want to make that change in the `sample-browser` as well.

Reply:
> I'm not sure what "tracking history without a browser fragment/ui around" means

We are able to load URLs while the app is in the background and while there may not be any UI (but we still want to record history). Fragments come and go. There's no reason we want to re-register a new delegate whenever there's a new `BrowserFragment` getting created.